tests/unit/utils/converter_unittest.cc

```diff
@@ -0,0 +1,84 @@
+#include "sg_gtest_utilities.h"
+
+#include <gtest/gtest.h>
+#include <shogun/lib/sg_types.h>
+#include <shogun/util/converters.h>
+
+#include <cstdint>
+#include <iostream>
+#include <limits>
+
+using namespace shogun;
+
+template <typename T>
+class Converters : public ::testing::Test
+{
+};
+
+SG_TYPED_TEST_CASE(Converters, sg_non_complex_types);
+
+TYPED_TEST(Converters, canDoCorrectConvertionOfUnsigned)
+{
+	TypeParam result = 0;
+	uint64_t value = 120ull;
+
+	ASSERT_NO_THROW(result = utils::safe_convert<TypeParam>(value));
+	EXPECT_EQ(static_cast<TypeParam>(value), result);
+}
+
+TYPED_TEST(Converters, canDoCorrectConvertionOfSigned)
+{
+	TypeParam result = 0;
+	int64_t value = 120ull;
+
+	ASSERT_NO_THROW(result = utils::safe_convert<TypeParam>(value));
+	EXPECT_EQ(static_cast<TypeParam>(value), result);
+}
+
+TYPED_TEST(Converters, cannotConvertOutOfRange)
+{
+	auto isLongDouble = typeid(TypeParam) == typeid(long double);
+
+	if (isLongDouble)
+	{
+		std::cout << "Disaled for long double since it fails" << std::endl;
```

Comment:
this could be only a comment only no need to write it out and just quietly ignore this test. i'm just wondering why the test fails for long?

Reply:
Got it. The test is green after applying the fix I mentioned above.